templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -0,0 +1,62 @@
+mode: 0755
+path: "/usr/local/bin/configure-ovs.sh"
+contents:
+  inline: |
+    #!/usr/bin/env bash
+    set -euxo pipefail
+
+    # create bridge
+    if ! nmcli connection show br-ex; then
+      brex_uuid=$(nmcli c add type ovs-bridge conn.interface br-ex con-name br-ex connection.autoconnect no | egrep '[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}' -o)
+    fi
+    if ! nmcli connection show ovs-port-br-ex; then
+      nmcli c add type ovs-port conn.interface br-ex master br-ex con-name ovs-port-br-ex connection.autoconnect no
+    fi
+
+    iface=""
+    counter=0
+    # find default interface
+    while [ $counter -lt 12 ]; do
+      if ip route | grep default; then
+        iface=$(ip route |grep default | grep -Eo "dev\s[^ ]+" | sed 's/dev\s//')
```

Comment:
In the event there are multiple interfaces connected to the same network, you could end up with multiple entries for the default gw.  Using my laptop as an example:

```
$ ip route | grep default
default via 192.168.1.1 dev enp0s31f6 proto dhcp metric 100 
default via 192.168.1.1 dev wlp58s0 proto dhcp metric 600 
$ ip route |grep default | grep -Eo "dev\s[^ ]+" | sed 's/dev\s//'
enp0s31f6
wlp58s0
```

Reply:
Probably want to pick the the interface with the lower metric after getting the ifaces

Reply:
nice catch, thanks

Reply:
looks like the routes are always sorted by lowest metric so I can just grab the first 1.
[root@test-q9gz9-bootstrap ~]# ip route
default via 192.168.126.1 dev br-ex 
default via 127.0.0.2 dev lo metric 50 
default via 192.168.126.1 dev br-ex metric 200 
default via 192.168.126.1 dev br-ex proto dhcp metric 800 
default via 192.168.126.1 dev br-ex metric 1299 
default via 127.0.0.2 dev lo metric 2000